libs/gl-client/src/credentials.rs

```diff
@@ -99,6 +103,22 @@ impl TlsConfigProvider for Nobody {
     fn tls_config(&self) -> TlsConfig {
         tls::TlsConfig::with(&self.cert, &self.key, &self.ca)
     }
+
+    /// Returns a new Nobody instance with a custom set of parameters.
+    /// Tries to load the CA from file and defaults to CA_RAW.
+    fn with_identity<V>(cert: V, key: V) -> Self
+    where
+        V: Into<Vec<u8>>,
+    {
+        let ca =
+            load_file_or_default("GL_CA_CRT", CA_RAW).expect("Could not load file from GL_CA_CRT");
+
+        Self {
+            cert: cert.into(),
+            key: key.into(),
+            ca,
+        }
+    }
```

Comment:
Is it really necessary to add this to the `TlsConfigProvider` trait? I fail to see the direct concern here. Maybe an extension or a separate trait would be cleaner?